### PR TITLE
Out-324 UI & Integration - Support attachments in Tasks

### PR DESCRIPTION
**Pull request by arpandhakal**
### Tasks

- [UI & Integration - Support attachments in Tasks](https://linear.app/copilotplatforms/issue/OUT-324/ui-and-integration-support-attachments-in-tasks)

### Changes

- [x] Applied supabase upload, fetch and remove attachments from client side.
- [x] Sent the path and relevant data coming from supabase to the server side. 
- [x] Integrated attachments api in the task details UI.
- [x] Added relavant icons to the icons list required for filetype in attachments.  

### Testing Criteria

- [LOOM](https://www.loom.com/share/7c8b7e7533b545aa88989c9e9251134a)

### Notes

- Dependent on OUT-325
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

### CODE REVIEW

**Review by rrojan — CHANGES_REQUESTED** [review]

File: src/lib/supabase.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import { supabaseAnonKey, supabaseProjectUrl } from '@/config'
+import { createClient } from '@supabase/supabase-js'
+
+export const supabase = createClient(supabaseProjectUrl, supabaseAnonKey)
```
[the remark this answers is not in the document]
Comment (arpandhakal) [reply src/lib/supabase.ts]: I think anon key meets our use case. ref : 
https://supabase.com/docs/guides/api/api-keys

File: src/utils/SupabaseActions.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  const { data, error } = await supabase.storage.from(supabaseBucket).download(filePath)
+  if (data) {
+    const url = window.URL.createObjectURL(new Blob([data]))
+    const link = document.createElement('a')
+    link.href = url
+    link.download = fileName
+    document.body.appendChild(link)
+    link.click()
+    document.body.removeChild(link)
+    window.URL.revokeObjectURL(url)
```
[the remark this answers is not in the document]
Comment (arpandhakal) [reply src/utils/SupabaseActions.ts]: URL is redirected from next router. We need to create a link for the blob url. 
ref : https://www.geeksforgeeks.org/how-to-implement-file-download-in-nextjs-using-an-api-route/

**Review by aatbip — APPROVED** [review]
lgtm

**Review by aatbip — DISMISSED** [review]
UI part lgtm - approved

File: src/app/api/attachments/attachments.controller.ts [hunk 3]
```diff
@@ -27,3 +36,14 @@ export const deleteAttachment = async (req: NextRequest, { params: { id } }: IdP
   await attachmentsService.deleteAttachment(id)
   return new NextResponse(null, { status: httpStatus.NO_CONTENT })
 }
+
+export const getSignedUrlUpload = async (req: NextRequest) => {
+  const fileName = req.nextUrl.searchParams.get('fileName')
+  if (!fileName) {
+    throw new APIError(httpStatus.BAD_REQUEST)
```
Comment (rrojan) [inline src/app/api/attachments/attachments.controller.ts:43]: Might be better to use a second message param for clarity - same for above

**Review by rrojan — APPROVED** [review]
Other than minor requested changes, LGTM :trophy:

**Pull request merged**